test/conftest.py

```diff
@@ -162,6 +158,21 @@ def no_san_server_with_different_commmon_name(
         yield cfg
 
 
+@pytest.fixture
+def san_proxy_with_server(
+    tmp_path_factory: pytest.TempPathFactory,
```

Comment:
Let's add `loopback_host` fixture here so we can get tests for IPv4 and IPv6 automatically as well. Check out the `san_server` fixture for an example.